Tests/SwiftDocCTests/Infrastructure/ReferenceResolverTests.swift

```diff
@@ -735,6 +736,117 @@ class ReferenceResolverTests: XCTestCase {
+            names: .init(title: "Something", navigator: nil, subHeading: nil, prose: nil),
+            pathComponents: ["Something"],
+            docComment: SymbolGraph.LineList(
+                inSourceComment.splitByNewlines.enumerated().map { lineOffset, line in
+                    SymbolGraph.LineList.Line(text: line, range: .init(
+                        start: .init(line: start.line + lineOffset, character: start.character),
+                        end: .init(line: start.line + lineOffset, character: start.character + line.count)
+                    ))
+                },
+                uri: sourceCodeURL.absoluteString // We want the "file://" prefix
+            ),
+            accessLevel: .public,
+            kind: .init(parsedIdentifier: .class, displayName: "Kind Display Name"),
+            mixins: [:]
+        )
+        
+        let (bundle, context) = try testBundleAndContext()
+        
+        let documentationExtensionContent = """
+        # ``Something``
+        
+        Continue the documentation for the "something" class.
+        
+        These other links to ``OtherNotFoundSymbol`` and <doc:OtherNotFoundArticle> also won't resolve.
+        
+        This other image name also won't resolve: ![Some other image that's not found](other-not-found-image)
+        """
+        let documentationExtensionURL = URL(fileURLWithPath: "/Users/username/path/to/SomeCatalog.docc/Something.md")
+        
+        var ignoredProblems = [Problem]()
+        let article = Article(
+            from: Document(parsing: documentationExtensionContent, source: documentationExtensionURL, options: [.parseSymbolLinks, .parseBlockDirectives]),
+            source: documentationExtensionURL,
+            for: bundle,
+            in: context,
+            problems: &ignoredProblems
+        )
+        XCTAssert(ignoredProblems.isEmpty, "Unexpected problems creating article")
+        
+        let node = DocumentationNode(
+            reference: reference,
+            symbol: symbol,
+            platformName: nil,
+            moduleReference: moduleReference,
+            article: article,
+            engine: context.diagnosticEngine
+        )
+        
+        XCTAssertEqual(node.docChunks.count, 2, "This node has content from both the in-source comment and the documentation extension file.")
+        
+        var resolver = ReferenceResolver(context: context, bundle: bundle)
+        _ = resolver.visitSymbol(node.semantic as! Symbol)
+        
+        let problems = resolver.problems.sorted(by: \.diagnostic.summary)
+        XCTAssertEqual(problems.count, 6)
+        
+        // These links to ``NotFoundSymbol`` and <doc:NotFoundArticle> won't resolve.
+        do {
+            let problem = try XCTUnwrap(problems.first)
+            XCTAssertEqual(problem.diagnostic.summary, "Can't resolve 'NotFoundArticle'")
+            XCTAssertEqual(problem.diagnostic.source?.path, "/Users/username/path/to/Something.swift")
```

Comment:
Can you test the diagnostic source ranges also here and below? Since this PR modified/moved the code that adjusts the offsets of the source lines, it would be good to test them here.

Reply:
Added in https://github.com/apple/swift-docc/pull/951/commits/e8bbd7063a2d27277d84ccae57ac19ea87c8fdeb. Note that the ReferenceResolver doesn't offset any diagnostics (neither before nor now). That happens in `DocumentationContext/resolveLinks(curatedReferences:bundle:)`.